examples/vae_text/vae_train.py

```diff
@@ -433,26 +401,52 @@ def _generate(filename: Optional[str] = None):
 
         fc_output = model.mlp_linear_layer(latent_z)
         if not isinstance(model.decoder_initial_state_size, torch.Size):
-            dcdr_states = torch.split(
+            decoder_states = torch.split(
                 fc_output, model.decoder_initial_state_size, dim=1)
         else:
-            dcdr_states = torch.reshape(
+            decoder_states = torch.reshape(
                 fc_output, (-1, ) + model.decoder_initial_state_size)
 
-        outputs = model.decode(
-            dcdr_states, latent_z, output_embed=None,
-            seq_lengths=None, is_generate=True)
+        if config.decoder_type == "lstm":
+            def _cat_embedder(ids):
+                """Concatenates latent variable to input word embeddings
+                """
+                embedding = model.decoder_w_embedder(
+                    ids.to(device))
+                return torch.cat([embedding, latent_z], 1)
```

Comment:
Based on the current implementation, can I assume you only concatenated the `latent_z`'s during generation but not training? This would be incorrect and result in poor performance.

Reply:
Got it, implemented in the `token_embedder`.